optimus_manager/xorg.py

```diff
@@ -168,11 +172,14 @@ def _generate_nvidia(config, bus_ids, xorg_extra):
 
     text += "EndSection\n\n"
 
-    text += "Section \"Device\"\n" \
-            "\tIdentifier \"integrated\"\n" \
-            "\tDriver \"modesetting\"\n"
-    text += "\tBusID \"%s\"\n" % bus_ids[integrated_gpu]
-    text += "EndSection\n\n"
+    if integrated_gpu == "intel":
+        text += "Section \"Device\"\n" \
+                "\tIdentifier \"integrated\"\n" \
+                "\tDriver \"modesetting\"\n"
+        text += "\tBusID \"%s\"\n" % bus_ids["intel"]
+        text += "EndSection\n\n"
+    else:
+        text += _make_amd_device_section(config, bus_ids, xorg_extra)
```

Comment:
Ok, this one is bothering me, but it's not your fault.

Currently, the config values in the `[amd]` section apply _only_ to "integrated" mode, and not "nvidia" mode ("nvidia" mode uses "modesetting" no matter what). Your change would make the configuration of the `[amd]` section apply in "nvidia" mode as well. But I agree that the previous behavior was confusing (and didn't allow to select `amdgpu`), so this change is actually for the best. I'll just have to add some comments in the config file to clarify which options apply to both modes.

Also, we have to apply the same change to Intel GPUs, for consistency. Could you replace the code block under the "if" branch by `text += _make_intel_device_section(config, bus_ids, xorg_extra)` ?